### ISSUE TITLE
Update metrics to use Enum

### ISSUE THREAD

**Issue by SkafteNicki** [PyTorchLightning/pytorch-lightning#5657]
### Motivation

Update metrics package to use `Enum` where it makes sense. For example:
https://github.com/PyTorchLightning/pytorch-lightning/blob/f78223041283450c98fb34bb23bea6706247d937/pytorch_lightning/metrics/classification/helpers.py#L79-L87
the `case` (data type) could be made an enum type
``` python
class DataType(LightningEnum):
    BINARY = 'binary'
    MULTILABEL = 'multi-label'
    MULTICLASS = 'multi-class'
    MDMC = 'multi-dim multi-class'
```

### Additional context

@Borda

**Comment by yuntai** [comment]
I could give it a shot.

something like this
```_check_shape_and_type_consistency(preds: torch.Tensor, target: torch.Tensor) -> Tuple[DataType, int]```
& typify all string values to DataType in the metric package?
can you point to a file where `DataType` should be defined?

**Comment by SkafteNicki** [comment]
The `DataType` is propably unique to classification problems and I would therefore put it in `pytorch-lightning/pytorch_lightning/metrics/classification/helpers.py` file.

**Comment by yuntai** [comment]
fine with 
```
class DataType(LightningEnum):
    BINARY = 'binary'
    MULTILABEL = 'multilabel'
    MULTICLASS = 'multiclass'
    MDMC = 'mdmc'
```
?

Otherwise
`DataType.from_str('multi-label') == None`
due to https://github.com/PyTorchLightning/pytorch-lightning/blob/84e02fbb4d082a8457b58464c5bd6631d9496693/pytorch_lightning/utilities/__init__.py#L58

**Comment by SkafteNicki** [comment]
Closed by PR #5689
